cast/java/ecj/src/main/java/com/ibm/wala/cast/java/translator/jdt/JDTJava2CAstTranslator.java

```diff
@@ -968,9 +1103,72 @@ public String toString() {
     return castAnnotations;
   }
 
-  protected final class ProcedureEntity
+  protected class ProcedureEntity
       implements JavaProcedureEntity { // TAGALONG (make static, access ast)
 
+    protected class JdtMethodCAstType implements CAstType.Method {
+      private Collection<CAstType> fExceptionTypes = null;
+
+      @Override
+      public boolean isStatic() {
+        return getQualifiers().contains(CAstQualifier.STATIC);
+      }
+
+      @Override
+      public CAstType getReturnType() {
+        if (fReturnType != null) return fTypeDict.getCAstTypeFor(fReturnType);
+        @SuppressWarnings("deprecation")
+        Type type =
+            fDecl == null
+                ? null
+                : (ast.apiLevel() == 2 ? fDecl.getReturnType() : fDecl.getReturnType2());
+        if (type == null) return fTypeDict.getCAstTypeFor(ast.resolveWellKnownType("void"));
+        else return fTypeDict.getCAstTypeFor(type.resolveBinding());
+      }
+
+      /** NOT INCLUDING first parameter 'this' (for non-static methods) */
+      @Override
+      public List<CAstType> getArgumentTypes() {
+        return fParameterTypes;
+      }
+
+      /** NOT INCLUDING first parameter 'this' (for non-static methods) */
+      @Override
+      public int getArgumentCount() {
+        return fDecl == null
+            ? fParameterTypes != null ? fParameterTypes.size() : 0
+            : fParameterTypes.size();
+      }
+
+      @Override
+      public String getName() {
+        Assertions.UNREACHABLE("CAstType.FunctionImpl#getName() called???");
```

Comment:
Nit: I think these days, rather than `Assertions.UNREACHABLE` we should just `throw new UnsupportedOperationException(msg)`; this way we won't have to write a `return` statement after.